src/components/Login.js

```diff
@@ -79,84 +74,99 @@ class Login extends React.Component {
           x1="547"
           y1="0"
           x2="547"
-          y2={`${window.innerHeight}`}
+          y2={`${windowHeight}`}
           gradientUnits="userSpaceOnUse"
         >
-          <stop stopColor={gradientColors[this.state.index][0]} />
-          <stop offset="1" stopColor={gradientColors[this.state.index][1]} stopOpacity="0.47" />
+          <stop stopColor={gradientPrimary} />
+          <stop offset="1" stopColor={gradientSecondary} stopOpacity="0.47" />
         </linearGradient>
       </defs>
     </svg>
   );
 
-  render = () => {
-    const textHighlightStyle = {
-      background: `linear-gradient(180deg, rgba(255,255,255,0) 80%, ${
-        gradientColors[this.state.index][0]
-      } 50%)`,
-    };
-    return (
-      <div className="login-page-content">
-        <div className="login-page-content-container">
-          <div
-            className="bottom-right-toggle"
-            onClick={() => this.setState({ index: Math.floor(Math.random() * 5) })}
-          >
-            <FontAwesomeIcon icon={faRedo} />
-          </div>
-          <div className="login-page-content-main">
-            <div>
-              <h1 className="font-weight-bold">
-                The ACM
-                {' '}
-                <span className="teachla-green">Teach LA</span>
-                {' '}
-                <span style={textHighlightStyle}>Online Editor</span>
-              </h1>
-              <p>a web IDE that lets you write and run Python &amp; Processing code, anywhere.</p>
-              {this.props.create ? (
-                <CreateUserForm
-                  initialState={this.props.initialState}
-                  themeColor={themeColors[this.state.index][0]}
-                  textColor={themeColors[this.state.index][2]}
-                />
-              ) : (
-                <LoginForm
-                  themeColor={themeColors[this.state.index][0]}
-                  textColor={themeColors[this.state.index][1]}
-                />
-              )}
-            </div>
-          </div>
-          <div className="login-page-content-footer">
-            <FontAwesomeIcon icon={faPaintBrush} />
-            {' '}
-            <FontAwesomeIcon icon={faCode} />
-            {' '}
-            <FontAwesomeIcon icon={faRocket} />
-            {' '}
-            by
-            {' '}
-            <a href="https://teachla.uclaacm.com" target="_blank" rel="noopener noreferrer">
-              <span className="teachla-green">ACM Teach LA</span>
-            </a>
-            {' '}
-            with
-            {' '}
-            <FontAwesomeIcon className="beating-heart" icon={faHeart} />
+  const [svg, setSvg] = useState(getBackgroundSVG());
+  const updateSvg = () => {
+    setSvg(getBackgroundSVG());
+  };
```

Comment:
I'm a bit confused as to why we need this state variable - don't all the state variables in `getBackgroundSVG` already rely on existing state, and thus automatically change on re-render? Right now, we're putting this JSX element into state, but it should be generated *from* state.

Reply:
When the svg is in state, the background svg is moving in an interesting wavy way that I thought was the intended effect but it's possible that it's a bug.  I'll remove the state and push up shortly